### ISSUE TITLE
Fast Save and Use Delegates causing crash during new game world generation

### ISSUE THREAD

**Issue by dk88711** [peterhaneve/ONIMods#142]
As title.

## Expected Behavior
No crash.

## Current Behavior
If "Use Delegates" option is enabled, during new game world generation, crash screen appear or world generation failed message pops up.

I can disable "Use Delegates" to create a new world and enable it afterwards to enter the same world no problem. Just cannot create a new world when "Use Delegates" option is enabled,

## Mod Versions
Mod version 3.2.0.0
Game build FA-472345-BD

## Steps to Reproduce
1. "Use Delegates" enabled.
2. New Game -> Start Game -> wait until it crash.

## Output Log
Log in crash screen as follow.
```
SolarSystem.Load Error!
 Tried to deserialize a class named: Klei.ClusterLayoutSave but no such class exists   at PeterHan.FastSave.FastSerializationManager.GetFastDeserializationMapping (System.Type type) [0x0001f] in <24dba23766d14606a00e67cd36c247ce>:0 
  at PeterHan.FastSave.FastSerializePatches+Deserializer_DeserializeTypeless_Long_Patch.Prefix (System.Type type, System.Object obj, IReader reader, System.Boolean& __result) [0x00000] in <24dba23766d14606a00e67cd36c247ce>:0 
  at (wrapper dynamic-method) KSerialization.Deserializer.KSerialization.Deserializer.DeserializeTypeless_Patch1(System.Type,object,IReader)
  at KSerialization.Deserializer.Deserialize (System.Object obj, IReader reader) [0x0001c] in <8f0726745db24471804d5705938c3f79>:0 
  at ProcGenGame.Cluster.Load () [0x00029] in <06f8e7422a7d4d098b0ee5ec07bdf81a>:0 

  at UnityEngine.Debug.LogError (System.Object message) [0x00000] in <9fe2cbebfa394c8da6af069ec8cc9ef4>:0 
  at Debug.LogError (System.Object obj) [0x00000] in <8f0726745db24471804d5705938c3f79>:0 
  at DebugUtil.LogErrorArgs (System.Object[] objs) [0x00000] in <8f0726745db24471804d5705938c3f79>:0 
  at ProcGenGame.Cluster.Load () [0x00000] in <06f8e7422a7d4d098b0ee5ec07bdf81a>:0 
  at SaveLoader.LoadFromWorldGen () [0x00000] in <06f8e7422a7d4d098b0ee5ec07bdf81a>:0 
  at SaveLoader.OnSpawn () [0x00000] in <06f8e7422a7d4d098b0ee5ec07bdf81a>:0 
  at KMonoBehaviour.Spawn () [0x00000] in <8f0726745db24471804d5705938c3f79>:0 
  at KMonoBehaviour.Start () [0x00000] in <8f0726745db24471804d5705938c3f79>:0 
Build: FA-472345-BD
```

**Comment by peterhaneve** [comment]
Use Delegates is an experimental feature. Use at your own risk!

This particular issue has been fixed already in 23d5c857, but this version has not yet made it to Steam due to the Mergedown's continued instability.

**Comment by peterhaneve** [comment]
Fast Save version 3.3 now includes a fix to Use Delegates. Use it with caution, as it may have further unknown bugs - but it can make a big difference in autosave performance on some colonies.
